src/Collection.php

```diff
@@ -1517,4 +1517,64 @@ public function stats()
             'collstats' => $this->getName(),
         ));
     }
+
+    /**
+     * Rename collection x to y
+     * Note: if collection existent, then command will fail
+     *
+     * @param  string $target   The name of target db and collection ex.: "testdb.newcollection"
+     * @return \MongoCollection
+     * @throws \Sokil\Mongo\Exception
+     */
+    public function renameCollection($target)
+    {
+        return $this->replaceRenameCollection($target, false);
+    }
+
+    /**
+     * Rename collection x to y, with dropping previously created target
+     *
+     * @param  string $target   The name of target db and collection ex.: "testdb.newcollection"
+     * @return \MongoCollection
+     * @throws \Sokil\Mongo\Exception
+     */
+    public function replaceCollection($target)
+    {
+        return $this->replaceRenameCollection($target, true);
+    }
+
+    /**
+     * Rename collection x to y, with dropping previously created target or not optionally
+     *
+     * @param string $target        The name of target db and collection ex.: "testdb.newcollection"
+     * @param  bool  $dropTarget    Whether to drop target or not
+     * @return \MongoCollection
+     * @throws \Sokil\Mongo\Exception
+     */
+    private function replaceRenameCollection($target, $dropTarget)
+    {
+        // create admin db instance
+        $adminDb = $this->getDatabase()->getClient()->getDatabase('admin');
+
+        // rename current collection to target
+        $response = $adminDb->executeCommand(array(
+            'renameCollection' => $this->getDatabase()->getName() . '.' . $this->getName(),
+            'to'               => $target,
+            'dropTarget'       => $dropTarget,
+        ));
+
+        if ($response['ok'] === 1.0) { // renamed successfully
+            // re-init collection
+            list($db, $coll) = explode('.', $target);
```

Comment:
target may be defined without point. in this case we move collection to same db. But here you ll get notice

Reply:
I thought it will always be `dbname.collname` like semantic - will make it more flexible,

Reply:
then you need to check that point exists before trying to rename. but more clear way imho allow db name be optional